server/accounts.go

```diff
@@ -2623,6 +2632,29 @@ func (s *Server) UpdateAccountClaims(a *Account, ac *jwt.AccountClaims) {
 			}
 		}
 	}
+
+	if _, ok := s.incompleteAccExporterMap.Load(old.Name); ok {
+		s.incompleteAccExporterMap.Delete(old.Name)
+		s.accounts.Range(func(key, value interface{}) bool {
+			acc := value.(*Account)
+			acc.mu.RLock()
+			incomplete := acc.incomplete
+			// Must use jwt in account or risk failing on fetch
+			// This jwt may not be the same that caused exportingAcc to be in incompleteAccExporterMap
+			claimJWT := acc.claimJWT
+			acc.mu.RUnlock()
+			if incomplete {
+				if accClaims, _, err := s.verifyAccountClaims(claimJWT); err == nil {
+					s.UpdateAccountClaims(acc, accClaims)
```

Comment:
This is the function from which we are inside. So we will possibly recurse. Do you see any problem with that?

Reply:
Not really.
sync.Map has no problem with it. 
We are also only recursive for cases where the account is incomplete. 
This happens at the tail end of UpdateAccountClaims, so I'm not worried about not being fully set up.
In the recursive situation, an incomplete account can turn complete and thus not be considered any longer.
At any given time I believe the number of incomplete accounts to be small.
#inclomplete accounts * # accounts.

**I guess this comes back to: is ranging over all accounts appropriate at all?**
If there is one persistently bad account every time we update the exporter we will visit all accounts.
Say the token expired, because we don't get rid of the clients or the account, we will range over all accounts.

Reply:
Valid concern and - although we do range over accounts, for instance on reload - if we plan to have many, many accounts, that could become an issue. But in order to have an impact (that is sync.map range being slow) I am guessing that it would require a LOT of accounts, no?

Reply:
I would say we need a great many accounts. 
And then, this probably happens not that often as we remove the account from incompleteAccExporterMap.

I also want to point out that this scheme only works as long as we can continue to tolerate this situation.
update account A, fail to fetch import on account B. 
While account B has no clients, account A won't be updated.